### ISSUE TITLE
UI Redesign

### ISSUE THREAD

**Issue by mohammadrafigh** [Caleb9/downloader#2]
I'm going to redesign the UI so I opened this issue to track related tasks here.

- [x] Prepare a prototype
- [ ] Configure a framework
- [ ] Configure the UI library (Bulma)
- [ ] Implement the new design

**Comment by mohammadrafigh** [comment]
Take a look at this, what do you think?

The autocomplete shows a list of previously used sub directories, later we can just show available sub directories on disk.
By pressing "/" or "+" button, a chip would be added for sub directory which could be removed by pressing "X". We can also separate chips by "/" to show the directory hierarchy. Please let me know your thoughts or any questions you might have.

<img width="1440" alt="Screenshot 2023-08-22 at 11 01 26 PM" src="https://github.com/Caleb9/downloader/assets/6737508/2ef20f89-20b8-41f9-9050-4ccec1567d32">
<img width="1440" alt="Screenshot 2023-08-22 at 11 01 44 PM" src="https://github.com/Caleb9/downloader/assets/6737508/d16d30eb-b1f3-46b4-94f0-da95eadb8454">

**Comment by Caleb9** [comment]
Hi @mohammadrafigh, looks fantastic! :)

> The autocomplete shows a list of previously used sub directories, later we can just show available sub directories on disk.

Previously used sub-directories kept e.g. in a cookie is a fine idea, but showing what's there on the disk can create some problems. The back-end would need to support things like symbolic links, and the hierarchy can in extreme scenarios have bazzillions of entries :). What do you think to keep things simple for now?

> By pressing "/" or "+" button, a chip would be added for sub directory which could be removed by pressing "X". We can also separate chips by "/" to show the directory hierarchy. Please let me know your thoughts or any questions you might have.

I like the idea of separating by "/", it conveys better that it's a filesystem directory structure.

Also, because this looks like a complete rewrite of the front-end part, I'd like to ask if you preserved the functionality that detects target filename from the URL when not specified by the user?

Anyway, great job! :)

**Comment by mohammadrafigh** [comment]
Thank you. The chip is that rectangle below directory input which also has
a "X" (for example sub 1). Yes I won't touch that part. It's a prototype, I
didn't implement it yet. If you are agree with this design then let me know
so I will start working on it.

**Comment by Caleb9** [comment]
Yup, it looks great I think :). Please note that I edited my previous comment so there's a bit more there now than in the email notification you initially received and probably responded to.

**Comment by mohammadrafigh** [comment]
Oops! 😃 
Yes localstorage, and indexeddb are better since some people disable cookies. But we can also get directories from recently downloaded files (if any available). It's a good thing to have a starting point on a new device. For too much entries we can use pagination but I agree keep it simple for now, lets just do it client side. 

Sure I will keep both + and / and separate chips by /.

Well I will add a lightweight framework and start implementing it in a few days.
